### DISCUSSION TITLE
1.22.1

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#9931] [General]
# Notes
- For Home Assistant users only: because of the added `entity_category` support this release requires at least Home Assistant 2021.11.0
- For people using CC2530/CC2531 adapters: a new firmware has been released (`20211115` and `20211116`). This update is important for CC2530 users since it fixes an issue where the CC2530 has to be replugged when restarting Zigbee2MQTT. Link to [new firmwares](https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_Home_1.2/bin) and [changelog](https://github.com/Koenkk/Z-Stack-firmware/blob/cf267a08b74fdc4f5ed4c1f83a904967269f1a8c/coordinator/Z-Stack_Home_1.2/CHANGELOG.md).

# Improvements
- [#3269](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3269) Increase interval for pIHeatingDemand to save thermostat battery
- #9293 #9309 #9305 Add Home Assistant `entity_category` support
- #9312 Add support for unit of measurement to Home Assistant number entities
- [#3281](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3281) Expose `off` action for IKEA E1812
- [#3268](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3268) Normalise attribute reporting intervals
- [#9306](https://github.com/Koenkk/zigbee2mqtt/pull/9306) Expose `update_available` sensor with `device_class` `update` to Home Assistant
- [#3285](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3285) Improve Xiaomi WSDCGQ11LM pressure accuracy
- #9511 Show actual firmware version of device in Home Assistant instead of Zigbee2MQTT version
- [#3288](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3288) Change TuYa TS0601_rcbo  current measurement unit from `mA` to `A`
- [#446](https://github.com/Koenkk/zigbee-herdsman/pull/446) Check-in command support
- [#447](https://github.com/Koenkk/zigbee-herdsman/pull/447) Try to immediately send command to devices when `sendWhenActive` is used
- [#3308](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3308) Expose binary vibration sensor for Xiaomi DJT11LM
- [#3313](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3313) Support `power_on_behavior` for TuYa TS0115 
- [#9465](https://github.com/Koenkk/zigbee2mqtt/issues/9465) Support `switch_type` for TuYa TS0011_switch_module, TS0012_switch_module and TS0013_switch_module
- [#3323](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3323) Add cool system_mode for Zen-01-W
- [#9574](https://github.com/Koenkk/zigbee2mqtt/discussions/9574) Show all clusters in frontend (even ones not defined in zigbee-herdsman)
- [#3335](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3335) Separately identify TS0011_switch_module, TS0012_switch_module and TS0013_switch_module
-  [#3337](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3337) Support pulse_configuration, interface_mode and current_summation for Develco ZHEMI101
- [#3330](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3330) Improve Aurora Lighting AU-A1ZBR1GW and AU-A1ZBR2GW stability
- [#3341](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3341) Add min, max and step value for `local_temperature_calibration`
- [#3359](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3359) Expose battery voltage for Xiaomi GZCGQ01LM
- [#3362](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3362) Expose load for ELKO 4523430
- [#9759](https://github.com/Koenkk/zigbee2mqtt/pull/9759) Expose `pi_heating_demand` to Home Assistant
- [#9700](https://github.com/Koenkk/zigbee2mqtt/issues/9700) Log failed to ping as warning instead of error
- [#3401](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3401) Support more actions for Xiaomi QBKG26LM
- #9844 Add device class for Home Assistant update available sensor
- [#3423](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3423) Add low_battery and check_meter for Develco ZHEMI101 

# Fixes
- [#3253](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3253) Properly handle update available check failing for IKEA device
- [#3267](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3267) Fixed PECLS01 battery percentage incorrect
- [#3277](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3277) Fix `fan_state` and `fan_mode` for IKEA E2007
- [#9089](https://github.com/Koenkk/zigbee2mqtt/issues/9089) Fix `color_mode` key error in Home Assistant for Lidl HG06467
- #9490 Stop pinging device (availability feature) when it is removed
- [#3284](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3284) Add running state for TH1400ZB
- #9507 Fix invalid `frontend.host` default value
- #9012 Fix min/max value for Home Assistant number entity
- #9520 Fix external extensions not started
- #9547 Fix Home Assistant discovery message for `running_state` `fan_only`
- #7051 Fix Neo NAS-AB02B0 not working
- #9539 Fix Tuya TS0003 3 gang switch module recognised as Zemismart ZM-L03E-Z
- [#3217](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3217) Fix Home Assistant command topics for multi endpoint climate devices
- [#9599]( https://github.com/Koenkk/zigbee2mqtt/issues/9599) ZM350STW1TCF doesn't support color temp startup
- [#3325](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3325) Fix state for Javis JS-SLK2-ZB
- Fix startup blocked by onEvent handlers
- [#1010](https://github.com/Koenkk/zigbee2mqtt.io/pull/1010) Fix colortemp range for Philips 8719514301542
- [#3353](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3353) Fix warning mode for Bitron AV2010/24A
- #9658 Fix properties missing when debounce is used under some conditions 
- [#9723](https://github.com/Koenkk/zigbee2mqtt/issues/9723) Allow setting `boost_time` greater than 100 via Home Assistant
- [#3384](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3384) Fix power_outage_memory for Xiaomi QBKG40LM and QBKG41LM
- [#3389](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3389/files) Fix QBKG40LM operation mode
- #9779 Fix no color support for Miboxer FUT038Z
- #9797  Fix power measurements not working for some TS011F_plug
- #9803 Fix typos in settings shema
- #9646 Fix endpoint missing from state for 545D6102
- [#3412](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3412) Fix `valve_position` over reported by a factor of 10 for TS0601_thermostat_1
- #9519 Send only 1 MQTT message when last_seen is enabled and Zigbee message is received
- #9843 Fix synced color temperature out of range when setting color
- #9218 Fix unsupported device error logs on interactions with group entities
- #9822 Fix "Possible EventEmitter memory leak detected. 1001 drain listeners added to [Socket]" warning
- #9898 Fix ICZB-RM11S battery percentage divided by 2
- [#3429](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3429) Fix device power source when device has backup battery
- #7158 Fix not all actions being available for TuYa TS004F

# New supported devices
This release adds support for 71 devices:
- `046677803087` Philips Hue White ambiance Cher ceiling light
- `07502L` Immax 4 in 1 multi sensor
- `1740193P0` Philips Hue Lucca wall light
- `1741730V7` Philips Hue Lily outdoor spot light
- `1742330P7` Philips Hue Calla outdoor
- `1746547P7` Philips Hue White and color ambiance Daylo outdoor wall lamp
- `1746730V7` Philips Hue Lily outdoor spot light
- `2CH-ZG-BOX-RELAY` Envilar 2 channel box relay
- `368308_2010` Lidl Silvercrest rdiator valve with thermostat
- `3RSP019BZ` Third Reality Zigbee / BLE smart plug
- `4080148P7` Philips Hue Signe table light
- `4080148U9` Philips Hue White and color ambiance Signe table light
- `4080248U9` Philips Hue White and color ambiance Signe floor light
- `4090231P9` Philips Hue Liane
- `4256050-ZHAC` Centralite 3-Series smart dimming outlet
- `5047131P8` Philips Hue White ambiance Buckram single spotlight
- `5047230P6` Philips Hue White ambiance Buckram double spotlight
- `73741_LIGHTIFY` OSRAM LIGHTIFY RT5/6 LED
- `7A-SS-ZABC-H0` Leedarson 4-in-1-Sensor
- `81848` AduroSmart ERIA smart plug (with power measurements)
- `8719514302235` Philips Hue White Filament Bulb E14
- `8719514328242` Philips Hue white ambiance E27 800lm with Bluetooth
- `8719514344723` Philips Akari downlight
- `8719514392830` Philips Hue White Ambiance E27 filament screw globe
- `8A-SS-BA-H0` Leedarson Motion Sensor
- `9290024683` Philips Hue white ambiance A19 1100lm with Bluetooth
- `9290024689` Philips Hue white and color ambiance B22 1100lm
- `9290024717` Philips Hue white and color ambiance E26/A19 1600lm
- `929002471901` Philips Hue white ambiance E27 1600lm with Bluetooth
- `9290024785` Philips Hue White Ambinance G25 E26 Edison Filament Globe
- `9290024896` Philips Hue white and color ambiance E27
- `9290030211` Philips Hue white Candle bulb E14 bluetooth
- `9290030509` Philips Hue smart plug - EU
- `929003054301` Philips Hue White Ambiance Cher Pendant
- `9290030675` Philips Hue motion sensor
- `929003099101` Philips Hue white ambiance Aurelle rectangle panel light
- `AC201` OWON HVAC controller/IR blaster
- `CCB432` CASAIA Rail-Din relay and energy meter
- `DLKZMK12LM` Xiaomi Aqara single switch module T1 (without neutral). Doesn't work as a router and doesn't support power meter
- `E1F-N9G` Sengled Smart LED filament candle (E12)
- `FUT038Z` Miboxer RGBW LED controller
- `GL-D-006P` Gledopto Zigbee 6W anti-glare downlight RGB+CCT (pro)
- `GL-D-007P` Gledopto Zigbee 12W anti-glare downlight RGB+CCT (pro)
- `HG06463B` Lidl Livarno Lux E27 G95 filament bulb
- `HG07834A` Lidl Livarno Lux GU10 spot RGB
- `HG07834B` Lidl Livarno Lux E14 candle RGB
- `HG07834C` Lidl Livarno Lux E27 bulb RGB
- `JR-ZDS01` Girier 1 gang mini switch
- `JS-MC-SENSOR-ZB` JAVIS Microwave sensor
- `MCCGQ14LM` Xiaomi Aqara E1 door & window contact sensor
- `PECMS01` Perenio Motion sensor
- `SLR1c` Hive Heating thermostat
- `SLR2c` Hive Dual channel heating and hot water thermostat
- `SP 234` Innr Smart plug
- `THS317` OWON Temperature and humidity sensor
- `TI0001-curtain-switch` Livolo Zigbee curtain switch (can only read status, control does not work yet)
- `TS0011_switch_module` TuYa 1 gang switch module - (without neutral)
- `TS0012_switch_module` TuYa 2 gang switch module - (without neutral)
- `TS0013_switch_module` TuYa 3 gang switch module - (without neutral)
- `TS0601_illuminance_temperature_humidity_sensor` TuYa Illuminance, temperature & humidity sensor
- `TV02-Zigbee` TuYa Thermostat radiator valve
- `WS7` JetHome 3-ch battery discrete input module
- `WXKG16LM` Xiaomi Aqara wireless remote switch E1 (single rocker)
- `WXKG17LM` Xiaomi Aqara E1 double key wireless switch
- `XHS1-UE` Universal Electronics Inc Wireless digital pet resistant PIR detector
- `ZB-1026` RGB Genie Zigbee LED dimmer controller
- `ZBT-CCTLight-Candle0904` Leedarson LED E14 tunable white
- `ZBT-CCTLight-GLS0904` Leedarson LED E27 tunable white
- `ZBT-DIMLight-GLS0800` Leedarson LED E27 warm white
- `ZB_A60_RGBCW` Ajax Online Smart Zigbee pro 12W A60 RGBCW bulb
- `PMM-300Z1` ShinaSystem SiHAS energy monitor

<hr /><em>This discussion was created from the release <a href='https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.22.1'>1.22.1</a>.</em>

**Comment by HolGORE** [comment]
Hello, 
sorry i'm new to this topic. 
will there be an updated version for CC1352P2_CC2652P_launchpad_coordinator_20210708.zip this month too? 
thanks!

**Reply by Koenkk** [reply]
A new fw is already available here: https://github.com/Koenkk/Z-Stack-firmware/tree/develop/coordinator/Z-Stack_3.x.0/bin . It is not officially released because:
- I want to include a touchlink fix in it (Texas Instruments is looking into this)
- I want to add the possibility to set the transmit power > 5dbm (which ITEAD Sonoff is looking into)

**Comment by KartoffelToby** [comment]
i'll not create a duplicate. this is related to this relase #9942

**Comment by acaonweb** [comment]
To upgrade to 1.22.1 I have to upgrade firmware too? Or I can leave it as is?
Thanx in advance

**Reply by sjorge** [reply]
You don't need to upgrade the firmware to upgrade to 1.22.1

**Reply by acaonweb** [reply]
Thanks

**Comment by VVVerh** [comment]
Hello! I have some troubles with this update on my DIY_Zintercom (by xyzroe). Can't change Mode via home assistant entity. In older versions (1.21)  this option was in entities list.

**Reply by Koenkk** [reply]
@xyzroe  could you check this?

**Reply by xyzroe** [reply]
I've been using a custom entity to set the mode, since the days when mode selection didn't sneak into HA at all.
I have input_select and two automations for reading and setting the mode. Here is example https://pastebin.com/raw/3s8R8FB3

Just updated to the last version of z2m. Here is disabled entity to select mode. But after turning it on, it does not display the current mode and does not allow changing the Zintercom mode.
As I understand it is because of name of the modes. More precisely in first capital letters. 
Zintercom uses the names of the modes starting with capital letters, and in HA it is created input_select with lowercase letters.
